kitsune/users/views.py

```diff
@@ -326,13 +327,31 @@ def profile(request, template, username):
         raise Http404('No Profile matches the given query.')
 
     groups = user_profile.user.groups.all()
+
+    if user.profile.facebook:
+        # Get Facebook Username only from Facebook profile URL by using Facebook Graph API
+        # More about Facebook Graph API can be get from
+        # here: https://developers.facebook.com/docs/graph-api
+        facebook_graph_url = user.profile.facebook.replace(
+            'www.facebook.com', 'graph.facebook.com').replace(
+            'facebook.com', 'graph.facebook.com').replace(
+            'profile.php?id=', '')
+        facebook_profile_data = requests.get(facebook_graph_url).json()
+
+    try:
+        facebook_profile_data['username']
+        facebook_username = facebook_profile_data['username']
+    except:
```

Comment:
A better way to do all of this is simply `facebook_username = facebook_profile_data.get('username', '')`, which tries to get the value, and if it values returns a default value (`''` in this case).